sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -1162,6 +1163,131 @@ case class GpuArraysOverlap(left: Expression, right: Expression)
   }
 }
 
+case class GpuMapFromArrays(left: Expression, right: Expression) extends GpuBinaryExpression {
+
+  private val mapKeyDedupPolicy = SQLConf.get.getConf(SQLConf.MAP_KEY_DEDUP_POLICY)
+
+  override def dataType: MapType = {
+    MapType(
+      keyType = left.dataType.asInstanceOf[ArrayType].elementType,
+      valueType = right.dataType.asInstanceOf[ArrayType].elementType,
+      valueContainsNull = right.dataType.asInstanceOf[ArrayType].containsNull)
+  }
+
+  def compareOffsets(lhs: ColumnVector, rhs: ColumnVector) : Boolean = {
+    val boolScalar = withResource(lhs.getListOffsetsView) { lhsOffsets =>
+      withResource(rhs.getListOffsetsView) { rhsOffsets =>
+        withResource(lhsOffsets.equalTo(rhsOffsets)) { compareOffsets =>
+          compareOffsets.all
+        }
+      }
+    }
+    withResource(boolScalar) { all =>
+      all.getBoolean
+    }
+  }
+
+  def nullSanitize(lhs: ColumnVector, rhs: ColumnVector) :
```

Comment:
nit: Could we add some comments or explain a little bit better what exactly is happening here? It is not clear that we are dealing with nulls as the top level arrays.

Reply:
Updated!